### PR TITLE
build: update bazel to 3.4.1

### PR DESCRIPTION
**Pull request by rebello95**
Note that Bazel's toolchain for 3.4.1 is tagged 3.4.0: https://github.com/bazelbuild/bazel-toolchains/releases/tag/3.4.0.

Depends on https://github.com/envoyproxy/envoy-build-tools/pull/85.

Signed-off-by: Michael Rebello <me@michaelrebello.com>

Risk Level: Medium
Testing: CI
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -53,11 +53,11 @@ DEPENDENCY_REPOSITORIES = dict(
...
     ),
     bazel_toolchains = dict(
-        sha256 = "2431088b38fd8e2878db17e3c5babb431de9e5c52b6d8b509d3070fa279a5be2",
-        strip_prefix = "bazel-toolchains-3.3.1",
+        sha256 = "882fecfc88d3dc528f5c5681d95d730e213e39099abff2e637688a91a9619395",
+        strip_prefix = "bazel-toolchains-3.4.0",
         urls = [
-            "https://github.com/bazelbuild/bazel-toolchains/releases/download/3.3.1/bazel-toolchains-3.3.1.tar.gz",
-            "https://mirror.bazel.build/github.com/bazelbuild/bazel-toolchains/archive/3.3.1.tar.gz",
+            "https://github.com/bazelbuild/bazel-toolchains/releases/download/3.4.0/bazel-toolchains-3.4.0.tar.gz",
```
Comment (htuch) [inline bazel/repository_locations.bzl:59]: This is 3.4.0, did you want to bump all the way to 3.4.1?

Comment (rebello95) [reply bazel/repository_locations.bzl:59]: Please see the PR description 😄 

> Note that Bazel's toolchain for 3.4.1 is tagged 3.4.0: https://github.com/bazelbuild/bazel-toolchains/releases/tag/3.4.0

Comment (rebello95) [reply bazel/repository_locations.bzl:59]: Let me know if I missed something though!

**Comment by htuch** [comment]
The Windows failure looks legit. @wrowe @sunjayBhatia thoughts?

**Comment by sunjayBhatia** [comment]
@htuch envoy-build-tools would need to have a toolchain generated and committed for bazel 3.4.x, we can’t do docker in docker on windows so generating a toolchain on the fly doesn’t work

**Comment by sunjayBhatia** [comment]
See: https://github.com/envoyproxy/envoy-build-tools/blob/923df85a4ba7f30dcd0cb6b0c6d8d604f0e20f48/toolchains/regenerate.sh#L41

**Comment by rebello95** [comment]
Updating there: https://github.com/envoyproxy/envoy-build-tools/pull/85

**Pull request merged**